cryptography/src/secp256r1.rs

```diff
@@ -0,0 +1,672 @@
+//! secp256r1 implementation of the `Scheme` trait.
+
+use crate::{PrivateKey, PublicKey, Scheme, Signature};
+use commonware_utils::union_unique;
+use p256::{
+    ecdsa::{
+        signature::{Signer, Verifier},
+        SigningKey, VerifyingKey,
+    },
+    elliptic_curve::scalar::IsHigh,
+};
+use rand::{CryptoRng, Rng, SeedableRng};
+use std::borrow::Cow;
+
+const PUBLIC_KEY_LENGTH: usize = 33;
+const SIGNATURE_LENGTH: usize = 64;
+
+#[derive(Clone)]
+pub struct Secp256r1 {
+    signing_key: SigningKey,
+    verifying_key: VerifyingKey,
+}
+
+impl Scheme for Secp256r1 {
+    fn new<R: CryptoRng + Rng>(r: &mut R) -> Self {
+        let signing_key = SigningKey::random(r);
+        let verifying_key = signing_key.verifying_key().to_owned();
+        Self {
+            signing_key,
+            verifying_key,
+        }
+    }
+
+    fn from(private_key: PrivateKey) -> Option<Self> {
+        let signing_key = match SigningKey::from_slice(private_key.as_ref()) {
+            Ok(key) => key,
+            Err(_) => return None,
+        };
+        let verifying_key = signing_key.verifying_key().to_owned();
+        Some(Self {
+            signing_key,
+            verifying_key,
+        })
+    }
+
+    fn from_seed(seed: u64) -> Self {
```

Comment:
I wonder if there is a better way to reuse this code across implementations? We basically use the same implementation everywhere.

Reply:
A default trait function implementation could be added to avoid repeating the same logic everywhere.
It can be done in this PR or another one.

Reply:
I think it probably makes sense to do elsewhere.